### PR TITLE
Fix rust IR function/label parsing

### PR DESCRIPTION
**Pull request by OfekShilon**
Fixes at least #6554.

Modified to not clash with @narpfel 's earlier fix to #6558

### CODE REVIEW

File: lib/demangler/llvm.ts [hunk 1]
```diff
@@ -32,9 +32,8 @@ import {BaseDemangler} from './base.js';
 import {PrefixTree} from './prefix-tree.js';
 
 export class LLVMIRDemangler extends BaseDemangler {
-    // Identifiers can be quoted: https://llvm.org/docs/LangRef.html#identifiers
-    llvmSymbolRE = /@(?<symbol>[\w$.]+)/gi;
-    llvmQuotedSymbolRE = /@"(?<symbol>[^"]+)"/gi;
+
+    llvmSymbolRE = /@"?([^\s"(]+)/g;
```
Comment (OfekShilon) [inline lib/demangler/llvm.ts:36]: I had this fix stashed during a week of vacation, and I see now a different fix was merged. 
Maybe it's better to use a single regex, but *definitely* the quotes must be a part of the capture.

File: lib/cfg/cfg-parsers/llvm-ir.ts [hunk 2]
```diff
@@ -46,9 +46,9 @@ export class LlvmIrCfgParser extends BaseCFGParser {
 
     constructor(instructionSetInfo: BaseInstructionSetInfo) {
         super(instructionSetInfo);
-        this.functionDefinition = /^define .+ @(.+)\([^(]+$/;
```
Comment (OfekShilon) [inline lib/cfg/cfg-parsers/llvm-ir.ts:49]: This fails eg for
```
define internal fastcc void @"_ZN114_$LT$core..iter..adapters..flatten..FlatMap$LT$I$C$U$C$F$GT$$u20$as$u20$core..iter..traits..iterator..Iterator$GT$4next17h6bfdef8d5f43e5cfE"(ptr noalias nocapture noundef writeonly align 8 dereferenceable(24) %_0, ptr noalias noundef align 8 dereferenceable(128) %self) unnamed_addr personality ptr @rust_eh_personality {
```
And sadly the same regex also needs to work on the demangled version which contains not just quotes, but *spaces*:
```
define internal fastcc void @"<core::iter::adapters::flatten::FlatMap<I,U,F> as core::iter::traits::iterator::Iterator>::next"(ptr noalias nocapture noundef writeonly align 8 dereferenceable(24) %_0, ptr noalias noundef align 8 dereferenceable(128) %self) unnamed_addr personality ptr @rust_eh_personality {
```

File: lib/cfg/cfg-parsers/llvm-ir.ts [hunk 3]
```diff
@@ -46,9 +46,9 @@ export class LlvmIrCfgParser extends BaseCFGParser {
...
-        this.labelRe = /^([\w.-]+):\s*(;.*)?$/;
```
Comment (OfekShilon) [inline lib/cfg/cfg-parsers/llvm-ir.ts:50]: This failed eg for
```
"_ZN4core3ptr70drop_in_place$LT$core..option..Option$LT$alloc..string..String$GT$$GT$17hac30cc57262c391bE.exit.i":
```

File: lib/cfg/cfg-parsers/llvm-ir.ts [hunk 4]
```diff
@@ -46,9 +46,9 @@ export class LlvmIrCfgParser extends BaseCFGParser {
...
-        this.labelReference = /%([\w.-]+)/g;
```
Comment (OfekShilon) [inline lib/cfg/cfg-parsers/llvm-ir.ts:51]: This failed eg for:
```
%"_ZN4core3ptr70drop_in_place$LT$core..option..Option$LT$alloc..string..String$GT$$GT$17hac30cc57262c391bE.exit.i"
```
Note not just the quotes, but also `$`

**Comment by OfekShilon** [comment]
I couldn't write a comprehensive test for this since rust includes (randomly generated in CE) file names in IR types.
Here's one repro for the rust-ir-CFG problems, that this solves:  https://godbolt.org/z/x9jfc8WWa

**Comment by OfekShilon** [comment]
I really should add a 'wrap' button to IR. In rust the lines are getting out of hand.

**Review by jeremy-rifkin — APPROVED** [review]
I think this LGTM, thanks for the context on failed examples!

**Pull request merged**